DB/src/main/java/io/deephaven/db/v2/select/SelectFilter.java

```diff
@@ -181,32 +186,126 @@ public PreviousFilteringNotSupported(String message) {
         }
 
         @Override
         public void visit(FilterCondition match) {
-            out = ConditionFilter.createConditionFilter(Strings.of(match));
+            out = FilterConditionAdapter.of(inverted ? match.invert() : match);
         }
 
         @Override
         public void visit(FilterNot not) {
-            out = ConditionFilter.createConditionFilter(Strings.of(not));
+            out = not.filter().walk(new Adapter(!inverted)).getOut();
         }
 
         @Override
         public void visit(FilterIsNull isNull) {
-            out = ConditionFilter.createConditionFilter(Strings.of(isNull));
+            if (inverted) {
+                out = isNotNull(isNull.column());
+            } else {
+                out = isNull(isNull.column());
+            }
+        }
+
+        @Override
+        public void visit(FilterIsNotNull isNotNull) {
+            if (inverted) {
+                out = isNull(isNotNull.column());
+            } else {
+                out = isNotNull(isNotNull.column());
+            }
         }
 
         @Override
         public void visit(RawString rawString) {
-            out = SelectFilterFactory.getExpression(rawString.value());
+            if (inverted) {
+                out = SelectFilterFactory.getExpression(String.format("!(%s)", rawString.value()));
+            } else {
+                out = SelectFilterFactory.getExpression(rawString.value());
+            }
+        }
+
+        private static MatchFilter isNull(ColumnName columnName) {
+            return new MatchFilter(columnName.name(), new Object[] { null });
+        }
+
+        private static MatchFilter isNotNull(ColumnName columnName) {
+            return new MatchFilter(MatchType.Inverted, columnName.name(), new Object[] { null });
+        }
+
+        private static class FilterConditionAdapter implements Value.Visitor {
+
+            public static SelectFilter of(FilterCondition condition) {
+                FilterCondition preferred = condition.maybeTranspose();
+                return preferred.lhs().walk(new FilterConditionAdapter(condition, preferred)).getOut();
+            }
+
+            private final FilterCondition original;
+            private final FilterCondition preferred;
+
+            private SelectFilter out;
+
+            private FilterConditionAdapter(FilterCondition original, FilterCondition preferred) {
+                this.original = Objects.requireNonNull(original);
+                this.preferred = Objects.requireNonNull(preferred);
+            }
+
+            public SelectFilter getOut() {
+                return Objects.requireNonNull(out);
+            }
+
+            @Override
+            public void visit(ColumnName lhs) {
+                preferred.rhs().walk(new Visitor() {
+                    @Override
+                    public void visit(ColumnName rhs) {
+                        out = SelectFilterFactory.getExpression(Strings.of(original));
```

Comment:
You could just construct a `SwitchColumn`.

Reply:
SwitchColumn implements SelectColumn
not SelectFilter
just double checking there isn't a better SelectFilter for ColumnName=ColumnName

Reply:
I was wrong. I made https://github.com/deephaven/deephaven-core/issues/849 for the missing functionality.